src/components/Synthetics/MarketTokenSelector/MarketTokenSelector.tsx

```diff
@@ -0,0 +1,179 @@
+      return marketInfo.name.toLowerCase().indexOf(searchKeyword.toLowerCase()) > -1;
+    });
+  }, [marketsInfoData, searchKeyword, sortedMarketsByIndexToken]);
+
+  function handleSelectToken(marketTokenAddress: string) {
+    history.push({
+      pathname: "/pools",
+      search: `?market=${marketTokenAddress}`,
+    });
+  }
+
+  return (
+    <Popover className="MarketTokenSelector">
+      {({ open, close }) => {
+        if (!open && searchKeyword.length > 0) setSearchKeyword("");
+        return (
+          <div>
+            <Popover.Button as="div">
+              <button className={cx("chart-token-selector")}>
+                <span className="chart-token-selector--current inline-items-center">
+                  {currentMarketInfo && (
+                    <>
+                      <TokenIcon
+                        className="chart-token-current-icon"
+                        symbol={currentMarketInfo.isSpotOnly ? "swap" : currentMarketInfo.indexToken.symbol}
+                        displaySize={30}
+                        importSize={40}
+                      />
+                      <div className="Market-index-name">
+                        <div className="items-center">
+                          <span>GM{indexName && `: ${indexName}`}</span>
+                          <span className="subtext">{poolName && `[${poolName}]`}</span>
+                        </div>
+                        <div className="Market-subtext">GMX Market Tokens</div>
+                      </div>
+                    </>
+                  )}
+                </span>
+                <FaChevronDown fontSize={14} />
+              </button>
+            </Popover.Button>
+            <div className="chart-token-menu">
+              <Popover.Panel as="div" className="menu-items chart-token-menu-items">
+                <SearchInput
+                  className="m-md"
+                  value={searchKeyword}
+                  setValue={({ target }) => setSearchKeyword(target.value)}
+                  onKeyDown={(e) => {
+                    if (e.key === "Enter" && filteredTokens.length > 0) {
+                      handleSelectToken(filteredTokens[0].address);
+                      close();
+                    }
+                  }}
+                  placeholder="Search Market"
+                />
+                <div className="divider" />
+                <div className="chart-token-list">
+                  <table>
+                    {sortedMarketsByIndexToken.length > 0 && (
+                      <thead className="table-head">
+                        <tr>
+                          <th>MARKET</th>
+                          <th>MINTABLE</th>
+                          <th>SELLABLE</th>
+                          <th>APR</th>
+                        </tr>
+                      </thead>
+                    )}
+                    <tbody>
+                      {filteredTokens.map((market) => {
```

Comment:
i think it's better to move the logic outside JSX

Reply:
Updated